@xen-orchestra/web-core/lib/components/UiChip.vue

```diff
@@ -0,0 +1,181 @@
+<template>
+  <span class="ui-chip typo p3-regular" :class="[color, { disabled }]">
+    <span class="content" @click="emit('edit')">
+      <slot name="icon">
+        <UiIcon v-if="icon" :icon class="icon" fixed-width />
+      </slot>
+      <slot />
+    </span>
+    <UiIcon :icon="faCircleXmark" class="remove" @click.stop="emit('remove')" />
+  </span>
+</template>
+
+<script lang="ts" setup>
+import UiIcon from '@core/components/icon/UiIcon.vue'
+import type { Color } from '@core/types/color.type'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { faCircleXmark } from '@fortawesome/free-solid-svg-icons'
+
+withDefaults(
+  defineProps<{
+    icon?: IconDefinition
+    color?: Color
+    disabled?: boolean
+  }>(),
+  { color: 'info' }
+)
+
+const emit = defineEmits<{
+  (event: 'edit'): void
+  (event: 'remove'): void
+}>()
+</script>
+
+<style lang="postcss" scoped>
+/* COLOR VARIANTS */
```

Comment:
I've tried various things to avoid such repetition for an icon that has the same color 75% of the time.

I think we could handle it this way:

```suggestion
/* ICON COLOR VARIANTS */
.ui-chip {
  --icon-color: var(--color-grey-200);

  &:is(:disabled, .disabled) {
    --icon-color: var(--color-grey-600);

    &.info {
      --icon-color: var(--color-grey-400);
    }
  }
}

/* CHIP COLOR VARIANTS */
```